src/main/java/com/microfocus/application/automation/tools/run/SonarOctaneListener.java

```diff
@@ -161,31 +174,27 @@ private String getBuildNumber(Run<?, ?> run) {
 	public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
 	                    @Nonnull TaskListener listener) throws InterruptedException, IOException {
 		PrintStream logger = listener.getLogger();
-		ExtensionList<GlobalConfiguration> allConfigurations = GlobalConfiguration.all();
+		initializeSonarDetails(run,listener);
+
 		String jenkinsRoot = Jenkins.getInstance().getRootUrl();
 		String callbackWebHooksURL = jenkinsRoot + Webhooks.WEBHOOK_PATH + Webhooks.NOTIFY_METHOD;
-
-		if (run instanceof AbstractBuild) {
 			logger.println("callback URL for jenkins resource will be set to: " + callbackWebHooksURL);
-			String[] serverDetails = getSonarDetails(run, allConfigurations, listener);
 			OctaneSDK.getClients().forEach(octaneClient -> {
 				try {
-					octaneClient.getSonarService().ensureSonarWebhookExist(callbackWebHooksURL, serverDetails[0], serverDetails[1]);
+					octaneClient.getSonarService().ensureSonarWebhookExist(callbackWebHooksURL, getSonarServerUrl(), getSonarToken());
 				} catch (SonarIntegrationException e) {
 					logger.println("Web-hook registration in sonarQube for build " + getBuildNumber(run) + " failed: " + e.getMessage());
 				}
 			});
-			run.addAction(new WebhookAction(true, serverDetails[0],dataTypeList));
-
-		}
+		run.addAction(new WebhookAction(true, getSonarServerUrl(), dataTypeSet));
 	}
 
 	@Override
 	public SonarDescriptor getDescriptor() {
 		return (SonarDescriptor) super.getDescriptor();
 	}
 
-
+	@Symbol("addSonarQubeListener")
```

Comment:
The name of the pipeline will be "addSonarQubeListener" with no reference to Octane, wouldn't it be misleading to SonarQube Jenkins plugin?

Reply:
Changed to addALMOctaneSonarQubeListener